clang/lib/Parse/ParseDecl.cpp

```diff
@@ -4095,11 +4095,22 @@ void Parser::ParseDeclarationSpecifiers(
     case tok::kw_auto:
       if (getLangOpts().CPlusPlus11 || getLangOpts().C23) {
         if (isKnownToBeTypeSpecifier(GetLookAheadToken(1))) {
-          isInvalid = DS.SetStorageClassSpec(Actions, DeclSpec::SCS_auto, Loc,
-                                             PrevSpec, DiagID, Policy);
-          if (!isInvalid && !getLangOpts().C23)
-            Diag(Tok, diag::ext_auto_storage_class)
-              << FixItHint::CreateRemoval(DS.getStorageClassSpecLoc());
+          // In C++ (not C23), 'auto' cannot be combined with a type specifier.
+          // However, OpenCL has its own error handling for this case.
```

Comment:
What is the OpenCL handling? If it's the same as the general C++ handling, can we remove the OpenCL handling and enable the new code for OpenCL too?

Reply:
Yes, I think we can drop the OpenCL specific diagnostic in favor of the C++ diagnostic.